### PR TITLE
Convert AWS image data to schema approved format

### PR DESCRIPTION
**Pull request by FKolwa**
- Add new function 'format_image' to convert individual images according to JSON schema.
- Add new function 'format_all_images' to retrieve and convert all images for all regions.

Additional changes:
- Add missing architecture property to JSON schema.

### CODE REVIEW

File: src/rhelocator/update_images/aws.py [hunk 1]
```diff
@@ -114,3 +114,65 @@ def parse_image_name(image_name: str) -> dict[str, str]:
         return matches.groupdict()
 
     return {}
+
+
+def format_all_images() -> object:
```
Comment (FKolwa) [inline src/rhelocator/update_images/aws.py:119]: I don't like returning 'object' but the literal type annotation would be 'dict[str, dict[str, list[dict[str, str]]]]' ..
Anybody got a better idea?

Comment (major) [reply src/rhelocator/update_images/aws.py:119]: Haha, no better ideas here. This is where typing gets a little challenging.

File: src/rhelocator/update_images/aws.py [hunk 2]
```diff
@@ -114,3 +114,65 @@ def parse_image_name(image_name: str) -> dict[str, str]:
...
+    """Retrieve all AWS images from all regions and return a simplified data
+    representation.
+
+    Returns:
+        JSON like structure containting a list of streamlined image
+        information on all regions.
+    """
+    formated_images: list[dict[str, str]] = []
```
Comment (major) [inline src/rhelocator/update_images/aws.py:127]: Not a huge deal, but this should probably be `formatted_images` (two t's)

**Review by major — APPROVED** [review]
This looks good to me. It would be nice to get that `formated_images` fixed, but we could put it in a follow-on commit if that's easier to keep things moving along.

File: src/rhelocator/update_images/aws.py (hunk 2 above)
Comment (FKolwa) [reply src/rhelocator/update_images/aws.py:127]: "The fault, dear Brutus, is not in our <disabled linter>, but in our <ability to write plain English>"

**Pull request merged**